voyager_archive/views_utils.py

```diff
@@ -1,14 +1,36 @@
 from django.shortcuts import get_object_or_404
+from pymarc import Record
 from .models import AuthRecord, BibRecord, MfhdRecord
 
 
 def get_auth_record(auth_id: int) -> AuthRecord:
-    return get_object_or_404(AuthRecord, auth_id=auth_id)
+    rec = AuthRecord.objects.get(auth_id=auth_id)
+    marc_text = getattr(rec, "auth_record")
+    
+    byte_obj = marc_text.encode('utf-8')
```

Comment:
These 3 lines are the same in all three methods.  Please move to a new method, something like:
`def get_marc_fields(marc_text: str) -> list:`